pkgs/tools/system/pciutils/default.nix

```diff
@@ -14,8 +14,10 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [ pkg-config ];
-  buildInputs = [ zlib kmod which ] ++
-    lib.optional stdenv.hostPlatform.isDarwin IOKit;
+  buildInputs = [ zlib ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [ kmod ]
+    ++ [ which ]
+    ++ lib.optional stdenv.hostPlatform.isDarwin IOKit;
 
   preConfigure = if stdenv.cc.isGNU then null else ''
```

Comment:
```suggestion
  preConfigure = if (!stdenv.cc.isGNU) ''
```

Reply:
Is this valid syntax? Maybe I read suggestion incorrectly. I think `if` has to be a full expression:

```diff
-  preConfigure = if stdenv.cc.isGNU then null else ''
+  preConfigure = if (!stdenv.cc.isGNU) ''
     substituteInPlace Makefile --replace 'CC=$(CROSS_COMPILE)gcc' ""
   '';
```

Fails for me as:

```
$ nix build -f. pciutils
error: syntax error, unexpected ';', expecting THEN

       at /home/slyfox/dev/git/nixpkgs-staging-next/pkgs/tools/system/pciutils/default.nix:22:5:

           21|     substituteInPlace Makefile --replace 'CC=$(CROSS_COMPILE)gcc' ""
           22|   '';
             |     ^
           23|
```

Reply:
Obviously not. I forgot to include the lib.optionalString

Reply:
Ah, got it now. Added use of `lib.optionalString`.